woocommerce/woocommerce-gateway-stripe
# Add Bulgaria, Czech Republic, Greece, Cyprus, Malta, Slovenia

Fixes #1252 .

#### Changes proposed in this Pull Request:
- Adds Bulgaria, Czech Republic, Greece, Cyprus, Malta, Slovenia
- Removes Brazil, India and the Philippines. Stripe is not supported in the those countries. (Brazil and India are in preview, but not general availability.)
- Moves logic from payment request to main plugin
- Adds notice if the merchant's store is set to a country not supported by Stripe:

<img width="599" alt="Screen Shot 2020-07-15 at 3 23 01 PM" src="https://user-images.githubusercontent.com/1595739/87606137-1a3cf380-c6af-11ea-8a6d-e1c359f88c42.png">

#### Notes to Reviewers:
- In #628 @roykho initially implemented the country check for the payment request buttons only - see https://github.com/woocommerce/woocommerce-gateway-stripe/issues/628#issuecomment-386035477 - but as I re-read that thread I think it best to take the other proposed path and simply do not enable the Stripe extension for store countries that are not supported - that's how we do it for other plugins. In case there are installed users in those countries, they'll see the admin notice.

#### To Test:
- Ensure all automated tests pass `npm run test:php`
- Set the store to an supported country, e.g. the United States.
- Ensure that 1) no notice appears warning you that the country is unsupported, 2) that Stripe is shown in checkout and 3) that you can complete checkout.
- Set the store to an unsupported country, e.g. Brazil.
- Ensure that 1) a notice appears warning you about that and that 2) Stripe is not shown in checkout.

-------------------
